src/daemon/rpmostreed-os-experimental.cxx

```diff
@@ -299,11 +299,12 @@ rpmostreed_osexperimental_new (OstreeSysroot *sysroot, OstreeRepo *repo, const c
   g_assert (OSTREE_IS_SYSROOT (sysroot));
   g_assert (name != NULL);
 
-  g_autofree char *path = rpmostreed_generate_object_path (BASE_DBUS_PATH, name, NULL);
+  auto rust_name = rust::Str (name);
+  auto path = rpmostreecxx::generate_object_path (rust::Str (BASE_DBUS_PATH), { &rust_name, 1 });
```

Comment:
It looks like we have a minor problem here, mostly because the previous C logic was not properly handling error cases. This can throw an error-exception, so it should be wrapped in a `ROSCXX_TRY_VAL(...)` block.

Ideally it would be enough to add a `GError **error` to this function arguments, and wire it here. If some of the `gio` macros conflict with that, we'll have to workaround in some way.